core/controllers/contributor_dashboard.py

```diff
@@ -240,6 +241,89 @@ def _get_voiceover_opportunity_dicts(self, language_code, search_cursor):
         return opportunity_dicts, next_cursor, more
 
 
+class ReviewableOpportunitiesHandler(base.BaseHandler):
+    """Provides opportunities that have translation suggestions in review."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'topic_name': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            }
+        }
+    }
+
+    @acl_decorators.open_access
+    def get(self):
+        """Handles GET requests."""
+        topic_name = self.normalized_request.get('topic_name')
+        opportunity_dicts = [
+            opp.to_dict()
+            for opp in self
+                ._get_reviewable_exploration_opportunity_summaries(
+                    self.user_id, topic_name)]
+        self.values = {
+            'opportunities': opportunity_dicts,
+        }
+        self.render_json(self.values)
+
+    def _get_reviewable_exploration_opportunity_summaries(
+            self, user_id, topic_name):
+        """Returns exploration opportunity summaries that have translation
+        suggestions that are reviewable by the supplied user. The result is
+        sorted in descending order by topic, story, and story node order.
+
+        Args:
+            user_id: str. The user ID of the user for which to filter
+                translation suggestions.
+            topic_name: str. A topic name for which to filter the exploration
+                opportunity summaries. If 'All' is supplied, all available
+                exploration opportunity summaries will be returned.
+
+        Returns:
+            list(ExplorationOpportunitySummary). A list of the matching
+            exploration opportunity summaries.
+        """
+        # 1. Fetch the eligible topics.
+        # 2. Fetch the stories for the topics.
+        # 3. Get the reviewable translation suggestion target IDs for the user.
+        # 4. Get story exploration nodes in order, filtering for explorations
+        # that have in review translation suggestions.
+        if topic_name is None:
+            raise self.InvalidInputException
+
+        topics = []
+        if topic_name == feconf.ALL_LITERAL_CONSTANT:
+            topics = topic_fetchers.get_all_topics()
+        else:
+            topic = topic_fetchers.get_topic_by_name(topic_name)
+            if topic is None:
+                raise self.InvalidInputException(
+                    'The supplied input topic: %s is not valid' % topic_name)
+            topics = [topic]
+        exp_ids = []
+        in_review_suggestion_target_ids = (
+            suggestion_services
+            .get_reviewable_translation_suggestion_target_ids(user_id))
+        for topic in topics:
+            stories = story_fetchers.get_stories_by_ids([
```

Comment:
This seems like N fetches where N is the number of topics; could this get_multi() be made more efficient? (Ideally there is just 1 GET.)